File: src/main/scala/edu/austral/starship/scala/base/framework/GameManager.scala
package edu.austral.starship.scala.base.framework

import edu.austral.starship.scala.CustomGameFramework
import processing.core.PApplet
import processing.event.KeyEvent

import scala.collection.mutable

class GameManager extends PApplet {
  private val gameFramework: GameFramework = CustomGameFramework
  private var keySet = mutable.Set[Int]()

  override def settings(): Unit = gameFramework.setup(WindowSettings(this), ImageLoader(this))

  override def draw(): Unit = {
    clear()

    val timeSinceLastFrame = (frameRate / 60) * 100
    gameFramework.draw(g, timeSinceLastFrame, keySet.toSet)
  }

  override def keyPressed(event: KeyEvent): Unit = {
    keySet += event.getKeyCode

    gameFramework.keyPressed(event)
  }

  override def keyReleased(event: KeyEvent): Unit = {
    keySet -= event.getKeyCode

    gameFramework.keyReleased(event)
  }
}

File: src/main/java/edu/austral/starship/base/framework/GameManager.java
package edu.austral.starship.base.framework;

import edu.austral.starship.CustomGameFramework;
import processing.core.PApplet;
import processing.event.KeyEvent;

import java.util.HashSet;
import java.util.Set;

public class GameManager extends PApplet {
    private final GameFramework gameFramework = new CustomGameFramework();
    private final Set<Integer> keySet = new HashSet<>();

    public void settings() {
        gameFramework.setup(new WindowSettings(this), new ImageLoader(this));
    }

    public void draw() {
        clear();

        final float timeSinceLastFrame = (frameRate / 60) * 100;
        gameFramework.draw(g, timeSinceLastFrame, keySet);
    }

    public void keyPressed(KeyEvent event) {
        keySet.add(event.getKeyCode());

        gameFramework.keyPressed(event);
    }

    public void keyReleased(KeyEvent event) {
        keySet.remove(event.getKeyCode());

        gameFramework.keyReleased(event);
    }
}
